src/git/gitProviderService.ts

```diff
@@ -1334,6 +1334,28 @@ export class GitProviderService implements Disposable {
 		return provider.removeRemote(path, name);
 	}
 
+	@log()
+	async revert(repoPath: string | Uri, ref: string, flags: string[] | undefined = []): Promise<void> {
+		const { provider, path } = this.getProvider(repoPath);
+		if (provider.revert == null) throw new ProviderNotSupportedError(provider.descriptor.name);
+
+		const options: { edit?: boolean } = {};
+		for (const flag of flags) {
+			switch (flag) {
+				case '--edit':
+					options.edit = true;
+					break;
+				case '--no-edit':
+					options.edit = false;
+					break;
+				default:
+					break;
+			}
+		}
+
+		return provider.revert(path, ref, options);
+	}
```

Comment:
Yeah, we could parse that in `src/commands/git/*` directly. Since currently this method is only being called from commands I didn't have preference on where to perform that conversion, but we can absolutely do that before and send the options directly to the method. Even better, maybe we can have an alternative
```typescript
createFlagsQuickPickItem<Flags>(state.flags, ['--no-edit'], {
	label: this.title,
	description: '--no-edit',
	detail: `Will revert ${getReferenceLabel(state.references)}`,
}),
```

That accepts and returns an option set so we don't have to parse the flags every time. Let me think this through